### ISSUE TITLE
Docs structure

### ISSUE THREAD

**Issue by adamchainz** [django/asgiref#163]
I find it weird that asgiref's docs folder documents ASGI, and the `asgiref.sync` module is documented over in Django (3.1+ only).

I think it would be better to pull the docs for all asgiref functions over into asgiref's docs. The asgiref docs could then have two sections: ASGI, and the asgiref API.

**Comment by adamchainz** [comment]
(Intersphinx can be used from the Django docs to refer over to `asgi.readthedocs.io` smoothly)

**Comment by felixxm** [comment]
https://github.com/django/django/pull/11650#discussion_r390361047

**Comment by andrewgodwin** [comment]
Yes, this is mostly a legacy of the asgiref docs actually being the ASGI specification website. We should get around to fixing this.
